extensions/amp-sidebar/1.0/component.js

```diff
@@ -135,42 +137,88 @@ function SidebarWithRef(
   }, [opened, close]);
 
   return (
-    mounted && (
-      <>
-        <ContainWrapper
-          as={Comp}
-          ref={sidebarRef}
-          size={false}
-          layout={true}
-          paint={true}
-          part="sidebar"
-          wrapperClassName={`${classes.sidebar} ${
-            classes.defaultSidebarStyles
-          } ${side === Side.LEFT ? classes.left : classes.right}`}
-          role="menu"
-          tabindex="-1"
-          hidden={!side}
-          {...rest}
-        >
-          {children}
-        </ContainWrapper>
-        <div
-          ref={backdropRef}
-          onClick={() => close()}
-          part="backdrop"
-          style={backdropStyle}
-          className={`${backdropClassName ?? ''} ${classes.backdrop} ${
-            classes.defaultBackdropStyles
-          }`}
-          hidden={!side}
-        >
-          <div className={classes.backdropOverscrollBlocker}></div>
-        </div>
-      </>
-    )
+    <div className={objstr({[classes.unmounted]: !mounted})}>
+      <ContainWrapper
+        as={Comp}
+        ref={sidebarRef}
+        size={false}
+        layout={true}
+        paint={true}
+        part="sidebar"
+        wrapperClassName={objstr({
+          [classes.sidebar]: true,
+          [classes.defaultSidebarStyles]: true,
+          [classes.left]: side === Side.LEFT,
+          [classes.right]: side !== Side.LEFT,
+        })}
+        role="menu"
+        tabindex="-1"
+        hidden={!side}
+        {...rest}
+      >
+        {children}
+      </ContainWrapper>
+      <div
+        ref={backdropRef}
+        onClick={() => close()}
+        part="backdrop"
+        style={backdropStyle}
+        className={objstr({
+          [backdropClassName]: backdropClassName,
+          [classes.backdrop]: true,
+          [classes.defaultBackdropStyles]: true,
+        })}
```

Comment:
nit: [Place constant `true` values at the beginning](https://gist.github.com/alanorozco/4b7b9867d1b11c672939a8c3cb5d444e#-place-constant-true-values-at-the-beginning)

```suggestion
        className={objstr({
          [classes.backdrop]: true,
          [classes.defaultBackdropStyles]: true,
          [backdropClassName]: backdropClassName,
        })}
```

Also: if `backdrop` and `defaultBackdropStyles` don't occur independently, merge them.

Reply:
Thanks!  Updated to put `true` values at the top.  Regarding the `backdrop` and `defaultBackdropStyles`, the reason these are split out is because the `default` ones are meant to be overridable by the user, whereas the `backdrop` class has styles that are built in to the sidebar.  I sort of like having them split out to emphasize the distinction, but let me know if you feel strongly about it!

Reply:
I don't know if our JSS transform supports this (try it?) but how about exposing a single classname, but split them in definition anyhow?

```js
// these are meant to 
const defaultBackdropStyles = {
}

export {..., backdrop, ...};
```

Reply:
I don't know if our JSS transform supports this (try it?) but how about exposing a single classname, but split them in definition anyhow?

```js
// these can overridden
const defaultBackdropStyles = {
  ...
};

const backdrop = {
  // x: "... !important",
  ...defaultBackdropStyles,
};

export {..., backdrop, ...};
```

Reply:
also, in `backdrop`: shouldn't this be `!important`?

https://github.com/ampproject/amphtml/blob/343229a278f9a9de6db7457548a84cc39acf23cf/extensions/amp-sidebar/1.0/component.jss.js#L58

Reply:
Hmmm.. it doesn't seem to like it.  See below -

![image](https://user-images.githubusercontent.com/25626770/112371442-feb1d000-8cb4-11eb-8308-fd3a66f296fb.png)

You're right, I've moved zIndex to be configurable!

Reply:
I went ahead and merged for now, but let me know if you think there should be any other changes and I can follow up!